docs/service-bundle.md

```diff
@@ -0,0 +1,225 @@
+
+## Input
+
+The container is run with three arguments.
+
+* Action: a single word such as “provision”. The full list is below.
+* --extra-vars: This is ansible-specific and always has exactly the value
+  “--extra-vars”. An Ansible Playbook Bundle uses this and the third argument
+  as-is when calling ansible by appending them to the ansible command.
+* JSON: a json document of information that is useful to the bundle. Currently
+  it is expressed in a form that is useful for populating variables in ansible
+  that can be used by roles.
+
+### Actions
+
+Possible actions include
+* provision
+* deprovision
+* bind
+* unbind
+* update
+* test
+
+### JSON
+Example of JSON document passed to __provision__:
+
+```
+{
+    "cluster": "openshift",
+    "_apb_plan_id": "default",
+    "_apb_service_class_id": "c23ec213bb8dea1577230c5ce005b9c2",
+    "_apb_service_instance_id": "54636ad3-0378-49a1-a494-f97d0a0daf8e",
+    "mediawiki_admin_pass": "pass",
+    "mediawiki_admin_user": "admin",
+    "mediawiki_db_schema": "mediawiki",
+    "mediawiki_site_lang": "en",
+    "mediawiki_site_name": "MediaWiki",
+    "namespace": "bazproject"
+}
+```
+
+* cluster: the type of cluster on which the service bundle is being run
+  (currently either ‘openshift’ or ‘kubernetes’).
+* _apb_plan_id: the name of a “plan”, as defined through the OSB API, that is
+  available on the service class.
+* _apb_service_class_id: the “service_id”, as defined through the OSB API, that
+  is being used for the current action.
+* _apb_service_instance_id: the “instance_id”, as defined through the OSB API,
+  that uniquely identifies the service instance being acted upon.
+* namespace: the target namespace in which resources should be created,
+  deleted, or acted upon.
+
+Other keys represent parameter names from the Plan, the values for which have
+the type specified in the Plan.
+
+Example of JSON document passed to __bind__:
+
+```
+{
+    "_apb_plan_id": "dev",
+    "_apb_provision_creds": {
+        "DB_ADMIN_PASSWORD": "yiX2hXH7RU24RTVRXuye",
+        "DB_HOST": "postgresql",
+        "DB_NAME": "foo",
+        "DB_PASSWORD": "letmein",
+        "DB_PORT": "5432",
+        "DB_TYPE": "postgres",
+        "DB_USER": "bar"
+    },
+    "_apb_service_class_id": "1dda1477cace09730bd8ed7a6505607e",
+    "_apb_service_instance_id": "d6ac6b10-fbff-4944-8e6b-3478313e20d1",
+    "cluster": "kubernetes",
+    "namespace": "default"
+}
+```
+
+* _apb_provision_creds: A set of credentials and related data created during a
+  provision action for this service instance. Details are below in the Output
+  section.
+
+### Environment Variables
+
+The following environment variables are set by the broker:
+
+* __POD_NAMESPACE__: the namespace in which the service bundle is being run
+* __POD_NAME__: the name of the pod 
+
+## Output
+
+### stdout / stderr
+
+These are ignored.
+
+### Exit Code
+
+* 0 - success
+* 1 - error
+* 8 - action not implemented/supported
+
+### Encoded Credentials
```

Comment:
I think we should also have a section for last_operation since it's something we pass back to the broker.

Reply:
Just for clarity, `last_operation` is an annotation on the running "Service Bundle" pod that can be read from the broker to get progress information right?

Reply:
Yes.  The annotation on the pod is called ```apb_last_operation```.

Reply:
@rthallisey I don't think this document should read 'bundle.spec' because that would need to go through a proposal process. @mhrivnak is simply trying to document what it does today but use the terms "service bundle" in place of APB for the vernacular. Which leads to some confusing items as pointed out by @eriknelson, but it is where we are today. We will have to update this doc as we migrate.